views.py

```diff
@@ -1,45 +1,91 @@
 import random
 import datetime
+import logging
 
-from django.shortcuts import render_to_response
+from django.shortcuts import render
 from django.template import RequestContext
 
 from onisite.plugins.featured_content import config
 from core import models
 
 def featured(request):
-    number = config.NUMBER
-    pages = None
-    if config.THISDAY:
-        pages = _this_day()
-        if not pages is None:
-            this_day_title = True
-    if pages is None and config.RANDOM:
-        page_len = len(models.Page.objects.all())
-        select_pages = random.sample(xrange(1, page_len), number)
-        pages = map(_get_page_info_by_id, select_pages)
-    elif pages is None:
-        # grab randomly from the curated selection
-        page_info = config.PAGES
-        page_len = len(page_info)
-        if page_len <= number:
-            pages = map(_get_page_by_info, page_info)
-        else:
-            rand_nums = random.sample(range(1, page_len), number)
-            random_pages = []
-            for num in rand_nums:
-                random_pages.append(page_info[num])
-            pages = map(_get_page_by_info, random_pages)
-
-    return render_to_response('featured.html',
-                           dictionary=locals(),
-                           context_instance=RequestContext(request))   
+    # Seed the RNG with today's date so we always feature the same page(s) for
+    # an entire day
+    random.seed(datetime.date.today().strftime("%Y%m%d"))
+
+    # Get and randomize pages
+    pages = []
+    all_pages, this_day_title = _get_pages()
+    if len(all_pages) > 0:
+        rand_nums = random.sample(xrange(len(all_pages)), config.NUMBER)
+        for num in rand_nums:
+            pages.append(all_pages[num])
 
+    # Clear the seed so anything else using random numbers isn't affected
+    random.seed(None)
+
+    return render(request, 'featured.html', locals())
 
 # helper methods
 
-def _get_page_info_by_id(page_id):
-    page_obj = models.Page.objects.get(id=page_id)
+def _get_pages():
+    """Delegate page-fetching based on the configuration; if THISDAY is
+    requested and gets valid pages, the second argument returned is True to
+    signal to the template which title to use"""
+
+    # We use random pages if explicitly requested *or* THISDAY is requested but
```

Comment:
No time for love, Doctor Jones.  And by "love", I mean "style fixes".

To me, `is_rand` as a variable makes the "if" block clearer.  I think it's just a matter of preference, so unless you really fight it, I want to leave this as-is.